src/pybind/mgr/orchestrator/module.py

```diff
@@ -844,11 +844,21 @@ def _apply_misc(self, service_type=None, placement=None, unmanaged=False, inbuf=
   ceph orch apply -i <yaml spec>
   ceph orch apply <service_type> <placement> [--unmanaged]
         """
+        results = []
         if inbuf:
             if service_type or placement or unmanaged:
                 raise OrchestratorValidationError(usage)
             content: Iterator = yaml.load_all(inbuf)
-            specs = [ServiceSpec.from_json(s) for s in content]
+
+            specs = []
+            for s in content:
+                if 'service_type' in s and s['service_type'] == 'host':
+                    spec = HostSpec.from_json(s)
+                    spec.service_type = 'host'
+                    specs.insert(0, spec)
```

Comment:
changing the order here is a bit strange.

Reply:
hosts need to be at the front of the list of specs so they are added to the cluster before other specs that rely on them are applied. Ill try to find a way make this cleaner though